os/arch/xtensa/src/esp32/esp32_i2c.c

```diff
@@ -267,39 +286,88 @@ static int hsi2c_setup(struct esp32_i2c_priv_s *priv);
 static int i2c_hw_enable(i2c_port_t i2c_num);
 static int i2c_hw_disable(i2c_port_t i2c_num);
 static void i2c_hw_setpins(const i2c_config_t *config);
-
-uint32_t esp32_i2c_setclock(FAR struct i2c_dev_s *dev, uint32_t frequency);
-
+static void i2c_start_transfer(i2c_port_t i2c_num);
+static inline void i2c_esp32_reset_txfifo(i2c_port_t i2c_num);
+static void i2c_esp32_clear_commands(i2c_port_t i2c_num);
 static inline void i2c_esp32_reset_interrupts(i2c_port_t i2c_num);
+static void i2c_esp32_send_write_cmd(struct esp32_i2c_priv_s *priv, int addr_bytes, int cmd_start_index);
+static void i2c_esp32_send_read_cmd(struct esp32_i2c_priv_s *priv, int addr_bytes, int cmd_start_index);
+static void i2c_esp32_get_i2c_data(struct esp32_i2c_priv_s *priv);
+uint32_t esp32_i2c_setclock(FAR struct i2c_dev_s *dev, uint32_t frequency);
 
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+
+int i2c_set_timeout(i2c_port_t i2c_num, int timeout)
+{
+    I2C[i2c_num]->timeout.tout = timeout;
+    return 0;
+}
+
+int i2c_get_timeout(i2c_port_t i2c_num, int* timeout)
```

Comment:
Is i2c_set_timeout, i2c_get_timeout called by our code, if not, maybe now it is useless.

Reply:
Add two interfaces. Modify it to be static function. Thanks.